src/test/java/com/github/packageurl/PackageURLTest.java

```diff
@@ -173,7 +173,7 @@ void constructorParameters(String description, String purlString, String cpurlSt
         assertEquals(name, purl.getName());
         assertEquals(version, purl.getVersion());
         assertEquals(qualifiers, purl.getQualifiers());
-        assertEquals(subpath, purl.getSubpath());
+        //assertEquals(subpath, purl.getSubpath());
```

Comment:
Same as above.

Reply:
Actually, I was told that we were using the test suite incorrectly, and that we shouldn't compare the fields at all.

After berating me for even pointing out that there was an issue with the tests because these two tests introduced a different representation of the fields (original instead of canonical), they finally settled on changing the test suite so that all fields were original. At least there's an open pull request for that now.

So, either we'll end up having to remove everything except string comparison in our tests, or they'll finally settle on something. I am sort of against fixing the test suite locally in our copy, though, because we don't even know their intention.

Reply:
Actually, it was @jeremylong that submitted these tests upstream, so we can find out exactly what was intended with the fields. Was it a mistake or not? The tests don't pass with the current Java code and what do we do now about the fact that even after the "fix" in this pull request the Java code is failing the `subpath` comparison.

Reply:
Before continuing this one we may want to wait for the spec to be updated a bit - per comments like: https://github.com/package-url/purl-spec/pull/409#issuecomment-2727723076

I know at least one of my changes to the test case is being reverted due to a misunderstanding/wording of the spec.